pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/ComplexTypeTransformer.java

```diff
@@ -279,6 +294,23 @@ protected void flattenMap(GenericRow record, List<String> columns) {
     }
   }
 
+  /**
+   *
+   */
+  @VisibleForTesting
+  protected void dropPrefixes(GenericRow record) {
+    List<String> columns = new ArrayList<>(record.getFieldToValueMap().keySet());
```

Comment:
+1

Reply:
Good point. Since this was mainly intended for unnested fields, I was thinking it to be a rare edge case for prefixes matching regular columns since they'd lack the delimiter (ie "after." vs "after"), and thought it might help to have more flexibility to drop prefixes from regular columns if desired. I understand the concern though, so it can be updated it to drop prefixes only from columns generated by the ComplexTypeTransformer. 

Initially, I thought about doing this when generating the `flattenName`, but this would affect the `unnestCollection` logic here - https://github.com/apache/pinot/blob/master/pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/ComplexTypeTransformer.java#L154-L157 since we're flattening the map first (where we would have dropped prefixes) before unnesting any collections.